snuba/web/split.py

```diff
@@ -96,6 +218,11 @@ def execute(
         to_date = util.parse_datetime(to_date_str, date_align)
         from_date = util.parse_datetime(from_date_str, date_align)
 
+        if from_date != (from_date_literal.value if from_date_literal else None):
+            metrics.increment("mismatch.ast_from_date")
+        if to_date != (to_date_literal.value if to_date_literal else None):
+            metrics.increment("mismatch.ast_to_date")
```

Comment:
Are we expecting to see a number of mismatches here? I assume we'll have quite a few valid ones since we are checking all of the conditions now, not just the values passed in the "from_date" and "to_date" property.

Reply:
I hope not to see these discrepancies but at this point I don't think we can have enough data to tell whether the false positives would be a lot so I am planning to start collecting metrics and changing the approach according to the results.